### DISCUSSION TITLE
decentralized hub

### DISCUSSION THREAD

**Discussion by muhamadazmy** [threefoldtech/0-hub#62] [General]
I think this is becoming pressing issue by the minute. Right now ZOS nodes that lives away from europ can suffer from timeouts when trying to deploy VMs workloads (specially with full VMs) because it take too long to download a VM image over the hub.

Some issues where open on zos to tackle this problem like this one here https://github.com/threefoldtech/zos/issues/1921 but it might be the best solution for the problem.

We need to run multiple hubs in different locations that are mirror of each other (zdb sycn) or similar solutions. Another way is to do copy on request to local hubs when a file/chunck are requested and then they can be cached locally (in local hub) forever for faster access later.

**Comment by maxux** [comment]
Everything is in place right now to support master-slave replication (central existing hub and get copy of it). But supporting replication with different masters is not yet doable easily.
